test/test-fetch.sh

```diff
@@ -80,6 +80,13 @@ begin_test "fetch"
   assert_local_object "$contents_oid" 1
   assert_local_object "$b_oid" 1
 
+  # test with commit sha1 specified
+  rm -rf .git/lfs/objects
+  newbranch_sha1=$(git rev-parse newbranch)
+  git lfs fetch origin "$newbranch_sha1"
+  assert_local_object "$contents_oid" 1
+  assert_local_object "$b_oid" 1
```

Comment:
Do you think it may be helpful to add a commit to the remote that is a descendant of `$newbranch_sha1` and refute that the local object exists, just to make sure that this fetch doesn't just include everything?